src/storage-controller/src/lib.rs

```diff
@@ -2673,6 +2702,19 @@ where
             .collect();
 
         self.append_to_managed_collection(id, updates).await;
+
+        Some(
+            latest_row_per_id
+                .into_iter()
+                .filter_map(|(key, timestamp)| {
+                    match GlobalId::from_str(key.unwrap_str()) {
+                        Ok(id) => Some((id, timestamp.into())),
+                        // Ignore any rows that can't be unwrapped correctly
+                        Err(_) => None,
+                    }
+                })
+                .collect(),
+        )
```

Comment:
what about dropped aws privatelink connections that still have a history? for now its fine, but there should be a todo to add a `dropped` status dropped connections so that we can ignore them

Reply:
this also needs to be fixed for source/sink status histories as well, so we will need to come up with a shared impl for that